doc/source/cross-language.rst

```diff
@@ -0,0 +1,293 @@
+Cross-language programming
+==========================
+
+This page will show you how to use Ray's cross-language programming feature.
+
+Python calling Java
+-------------------
+
+Suppose we have a Java static method and a Java class as follows:
+
+.. code-block:: java
+
+  package io.ray.demo;
+
+  public class Math {
+
+    public static int add(int a, int b) {
+      return a + b;
+    }
+  }
+
+.. code-block:: java
+
+  package io.ray.demo;
+
+  // A regular Java class.
+  public class Counter {
+
+    private int value = 0;
+
+    public int increment() {
+      this.value += 1;
+      return this.value;
+    }
+  }
+
+Then, in Python, we can call the above Java remote function, or create an actor
+from the above Java class.
+
+.. code-block:: python
+
+  import ray
+
+  ray.init(_include_java=True, _load_code_from_local=True)
+
+  # Define a Java class.
+  counter_class = ray.java_actor_class(
+        "io.ray.demo.Counter")
+
+  # Create a Java actor and call actor method.
+  counter = counter_class.remote()
+  obj_ref1 = counter.increment.remote()
+  assert ray.get(obj_ref1) == 1
+  obj_ref2 = counter.increment.remote()
+  assert ray.get(obj_ref2) == 2
+
+  # Define a Java function.
+  add_function = ray.java_function(
+        "io.ray.demo.Math", "add")
+  
+  # Call the Java remote function.
+  obj_ref3 = add_function.remote(1, 2)
+  assert ray.get(obj_ref3) == 3
+
+  ray.shutdown()
+  
+Java calling Python
+-------------------
+
+Suppose we have a Python module as follows:
+
+.. code-block:: python
+
+  # ray_demo.py
+
+  class Counter(object):
+    def __init__(self):
+        self.value = 0
+
+    def increment(self):
+        self.value += 1
+        return self.value
+
+
+  def add(a, b):
+      return a + b
+
+Then, in Java, we can call the above Python remote function, or create an actor
+from the above Python class.
+
+.. code-block:: java
+
+  package io.ray.demo;
+
+  import io.ray.api.ObjectRef;
+  import io.ray.api.PyActorHandle;
+  import io.ray.api.Ray;
+  import io.ray.api.function.PyActorClass;
+  import io.ray.api.function.PyActorMethod;
+  import io.ray.api.function.PyFunction;
+  import org.testng.Assert;
+
+  public class JavaCallPythonDemo {
+
+    public static void main(String[] args) {
+      Ray.init();
+
+      // Define a Python class.
+      PyActorClass actorClass = PyActorClass.of(
+          "ray_demo", "Counter");
+
+      // Create a Python actor and call actor method.
+      PyActorHandle actor = Ray.actor(actorClass).remote();
+      ObjectRef objRef1 = actor.task(
+          PyActorMethod.of("increment", int.class)).remote();
+      Assert.assertEquals(objRef1.get(), 1);
```

Comment:
```suggestion
      Assert.assertTrue(objRef1.get() == 1);
```
Unfortunately, `assertEquals` doesn't support comparing `Integer` with `int.

Reply:
This case use `int.class`, so the `objRef1.get()` will return an `int` not an `Integer`.